system/CLI/Console.php

```diff
@@ -12,40 +12,28 @@
 namespace CodeIgniter\CLI;
 
 use CodeIgniter\CodeIgniter;
+use Config\Services;
 use Exception;
 
 /**
  * Console
  */
 class Console
 {
-    /**
-     * Main CodeIgniter instance.
-     *
-     * @var CodeIgniter
-     */
-    protected $app;
-
-    public function __construct(CodeIgniter $app)
-    {
-        $this->app = $app;
-    }
-
     /**
      * Runs the current command discovered on the CLI.
      *
      * @throws Exception
      *
      * @return mixed
      */
-    public function run(bool $useSafeOutput = false)
+    public function run()
     {
-        $path = CLI::getURI() ?: 'list';
-
-        // Set the path for the application to route to.
-        $this->app->setPath("ci{$path}");
+        $runner  = Services::commands();
+        $params  = array_merge(CLI::getSegments(), CLI::getOptions());
+        $command = array_shift($params) ?? 'list';
 
-        return $this->app->useSafeOutput($useSafeOutput)->run();
+        return $runner->run($command, $params);
```

Comment:
`BaseCommand::run()` called by `Commands::run()` returns `void` although not strictly because not typehinted. However, most of our implementations of commands I think don't return anything either. So the `$exit` in L85 in spark above will be `null` so always exit on `EXIT_SUCCESS`?

Reply:
It turns out that exit (0), unless of course the developer forcibly returns the desired value of the exit code.